src/btree/bt_discard.c

```diff
@@ -290,11 +290,8 @@ __wt_free_ref(WT_SESSION_IMPL *session, WT_REF *ref, int page_type, bool free_pa
     /* Free any address allocation. */
     __wt_ref_addr_free(session, ref);
 
-    /* Free any page-deleted information. */
-    if (ref->page_del != NULL) {
-        __wt_free(session, ref->page_del->update_list);
-        __wt_free(session, ref->page_del);
-    }
+    /* Free any backing fast-truncate memory. */
+    __wt_overwrite_and_free(session, ref->ft_info.del);
```

Comment:
Should there be an assertion here that ref->ft_info.update is NULL? I could be misunderstanding something.

Reply:
Well, it's a union now, so `ref.ft_info.del` is the same pointer as `ref.ft_info.update`. I thought about adding a `ft_info.both` field, to make it clear it was all going away, but didn't.

Reply:
Ah yes, okay 👍

Reply:
Sorry to reopen this one -- since `__wt_overwrite_and_free` uses `sizeof`, don't we need to know which pointer to use?

Reply:
@michaelcahill, nice catch -- thanks.

I've updated the calls where we don't know the pointer type, pushed in 36320998e.